egret/model_library/transmission/branch.py

```diff
@@ -776,6 +779,10 @@ def declare_ineq_p_branch_thermal_lbub(model, index_set,
     con_set = decl.declare_set('_con_ineq_p_branch_thermal_lbub',
                                model=model, index_set=index_set)
 
+    # flag for if slacks are on the model
+    pos_slacks = hasattr(model, 'pf_slack_pos')
+    neg_slacks = hasattr(model, 'pf_slack_neg')
```

Comment:
I would prefer it if we had an argument to the function which specified whether or not slacks should be used. If slacks are requested, but the variables do not exist, an exception should get raised. What do you think, @bknueven ?

Reply:
I think this works here. I've added the appropriate logic.